packages/forms/src/elements/number/number.tsx

```diff
@@ -44,22 +44,34 @@ const InputNumber: React.FC<InputNumberProps> = ({
 	};
 
 	const handleKeyDown = (e: any): void => {
-		e.target.value.length >= maxLength &&
-			!validKeys.includes(e.key) &&
-			e.preventDefault();
+		// avoid entering the number 'E'
 		e.key.toLowerCase() === 'e' && e.preventDefault();
 	};
 
+	const valueLengthValid = (value: string): boolean => {
+		// if maxLength specified, returns false when the length of the input is greater than maxLength
+		if (value.length > maxLength) {
+			return false;
+		}
+		return true;
+	};
+
 	const handleOnChange = (e: ChangeEvent<HTMLInputElement>): void => {
 		let newEvent = { ...e };
-		// if decimalPlaces && value not null => e.target.value = adaptValueToFormat(value, decimalPlaces)
-		// if decimalPlaces && value is null => e.target.value = parseInt(value)
-		decimalPlaces
-			? (newEvent.target.value =
-					e.target.value &&
-					adaptValueToFormat(newEvent.target.value, decimalPlaces))
-			: (newEvent.target.value =
-					e.target.value && parseInt(newEvent.target.value, 10).toString());
+		// the value is processed only when is a valid value
+		if (e.target.value !== '' && e.target.value !== '-') {
+			// if decimalPlaces && value not null => e.target.value = adaptValueToFormat(value, decimalPlaces)
```

Comment:
It would be good to see the old code removed here

Reply:
true, I forgot to update the old comments.